https://bugs.gentoo.org/580594
https://github.com/fujita/tgt/pull/25

From b092c6fe330a2eacf4b1d4eb093fad8e2fbcaed9 Mon Sep 17 00:00:00 2001
From: Mike Frysinger <vapier@gentoo.org>
Date: Sun, 27 Nov 2016 18:47:24 -0500
Subject: [PATCH] fix build w/newer glibc
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Building with newer glibc versions fails like so:
bs_sg.c: In function ‘chk_sg_device’:
bs_sg.c:354:6: error: implicit declaration of function ‘major’ [-Werror=implicit-function-declaration]
  if (major(st.st_rdev) == SCSI_GENERIC_MAJOR)

This is because glibc is dropping the implicit sys/sysmacros.h include
from sys/types.h and making the few projects that need it include it
explicitly.
---
 usr/bs_sg.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/usr/bs_sg.c b/usr/bs_sg.c
index 66f4a3b22a18..fb544056f258 100644
--- a/usr/bs_sg.c
+++ b/usr/bs_sg.c
@@ -32,6 +32,7 @@
 #include <linux/fs.h>
 #include <linux/major.h>
 #include <sys/ioctl.h>
+#include <sys/sysmacros.h>
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <sys/epoll.h>
-- 
2.11.0.rc2

